frontend/src/components/core/StreamlitDialog/StreamlitDialog.tsx

```diff
@@ -305,12 +305,28 @@ function uploadedDialog(props: UploadedProps): ReactElement {
     <BasicDialog onClose={props.onClose}>
       <ModalBody>
         <div className="streamlit-upload-first-line">App saved to:</div>
-        <div id="streamlit-upload-url"> {props.url} </div>
+        <div id="streamlit-upload-url">
+          <pre>
+            <a href={props.url} target="_blank" rel="noopener noreferrer">
+              {props.url}
+            </a>
+          </pre>{" "}
```

Comment:
Why is the space needed?

Reply:
That sounds reasonable, but this is the way that all the dialogs currently work. I'll open an issue to revise?

Reply:
This looks like a copy-paste mistake. I'll remove!